build-logic/convention/src/main/kotlin/AndroidApplicationComposeConventionPlugin.kt

```diff
@@ -0,0 +1,17 @@
+import com.android.build.gradle.internal.dsl.BaseAppModuleExtension
+import com.google.samples.apps.nowinandroid.configureAndroidCompose
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.getByType
+
+class AndroidApplicationComposeConventionPlugin : Plugin<Project> {
+    override fun apply(target: Project) {
+        with(target) {
+            pluginManager.withPlugin("com.android.application") {
```

Comment:
Yes, `withPlugin` is only running the action if the plugin is applied by the consumer project. 

> from [the doc](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/PluginManager.html#withPlugin-java.lang.String-org.gradle.api.Action-)
> `withPlugin​(String id, Action<? super AppliedPlugin> action)` | Executes the given action when the specified plugin is applied.

As for the ordering, I'm not sure I couldn't find anything in the documentation. I think it would be better to just apply the plugin instead 🤷

Reply:
I changed the behavior to be sure that the order doesn't matter.